accounts-db/src/tiered_storage/footer.rs

```diff
@@ -174,9 +174,33 @@ impl TieredStorageFooter {
         Self::new_from_footer_block(&file)
     }
 
-    pub fn write_footer_block(&self, file: &mut TieredWritableFile) -> TieredStorageResult<()> {
+    /// Presist the footer to the specified TieredStorageWritableFile.
+    /// The `hash` field of the footer will also be updated.
+    pub fn write_footer_block(&mut self, file: &mut TieredWritableFile) -> TieredStorageResult<()> {
         // SAFETY: The footer does not contain any uninitialized bytes.
-        unsafe { file.write_type(self)? };
+        unsafe {
+            file.write_type(&self.account_meta_format)?;
+            file.write_type(&self.owners_block_format)?;
+            file.write_type(&self.index_block_format)?;
+            file.write_type(&self.account_block_format)?;
+        }
+
+        file.write_pod(&self.account_entry_count)?;
+        file.write_pod(&self.account_meta_entry_size)?;
+        file.write_pod(&self.account_block_size)?;
+        file.write_pod(&self.owner_count)?;
+        file.write_pod(&self.owner_entry_size)?;
+        file.write_pod(&self.index_block_offset)?;
+        file.write_pod(&self.owners_block_offset)?;
+        file.write_pod(&self.min_account_address)?;
+        file.write_pod(&self.max_account_address)?;
+
+        // everything before the FooterTail will be hashed
+        self.hash = file.hash();
+        file.write_pod(&self.hash)?;
+
+        file.write_pod(&self.format_version)?;
+        file.write_pod(&self.footer_size)?;
```

Comment:
@brooksprumo: I think we can consider reordering the items in the footer tail so that the format version and footer-size are also hashed.

And, together with your previous comment https://github.com/anza-xyz/agave/pull/195#discussion_r1532496120.  If we also exclude hash or even footer tail from the footer, then we can directly do `file.write_type(footer)` here.

What do you think?

Reply:
Right!